src/ansible_compat/config.py

```diff
@@ -77,5 +84,50 @@ def ansible_version(version: str = "") -> Version:
     return Version(version)
 
 
+class AnsibleConfig(_UserDict):  # pylint: disable=too-many-ancestors
+    """Interface to query Ansible configuration.
+
+    This should allow user to access everything provided by `ansible-config dump` without having to parse the data himself.
+    """
+
+    _aliases = {
+        'COLLECTIONS_PATHS': 'COLLECTIONS_PATH',  # 2.9 -> 2.10+
+        'COLLECTIONS_PATH': 'COLLECTIONS_PATHS',  # 2.10+ -> 2.9
+    }
+
+    def __init__(self) -> None:
+        """Load config dictionary."""
+        super().__init__()
+        env = os.environ.copy()
+        # Avoid possible ANSI garbage
+        env["ANSIBLE_FORCE_COLOR"] = "0"
+
+        config = subprocess.check_output(
+            ["ansible-config", "dump"], universal_newlines=True, env=env
```

Comment:
How about doing this differently:
1. Have a class `AnsibleConfig` which takes the stdout of `ansible-config dump`.
2. Have a separate function that calls `ansible-config dump` and creates `AnsibleConfig` from the result.

This makes it possible to use this class in situations where you want to execute `ansible-config` in a specific venv, or want to execute it asynchronously (and not with `subprocess`).

Reply:
In fact I was planning to do this in follow-up, but now that I mentioned it I just did it. Note that I did not split yet the call to ansible, we will do this later with some king of ansible-runner class that will allow us to run various ansible commands, not only galaxy install.

Reply:
I'd also add that this "function" might as well be a `@classmethod` — it's a common tool for custom initializers.